x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,466 @@
+  region: string;
+  es_url: string;
+  kb_url: string;
+  product: string;
+}
+
+interface Credentials {
+  username: string;
+  password: string;
+}
+
+const DEFAULT_REGION = 'aws-eu-west-1';
+const PROJECT_NAME_PREFIX = 'kibana-cypress-security-solution-ephemeral';
+const BASE_ENV_URL = 'https://global.qa.cld.elstc.co';
+let log: ToolingLog;
+
+const getApiKeyFromElasticCloudJsonFile = (): string | undefined => {
+  const userHomeDir = os.homedir();
+  try {
+    const jsonString = fs.readFileSync(path.join(userHomeDir, '.elastic/cloud.json'), 'utf-8');
+    const jsonData = JSON.parse(jsonString);
+    return jsonData.api_key.qa;
+  } catch (e) {
+    log.info('API KEY could not be found in .elastic/cloud.json');
+  }
+};
+
+// Method to invoke the create environment API for serverless.
+async function createEnvironment(
+  projectName: string,
+  apiKey: string,
+  ftrConfig: SecuritySolutionDescribeBlockFtrConfig
+): Promise<Environment | undefined> {
+  const body: CreateEnvironmentRequestBody = {
+    name: projectName,
+    region_id: DEFAULT_REGION,
+  };
+
+  const productTypes: ProductType[] = [];
+  ftrConfig?.productTypes?.forEach((t) => {
+    productTypes.push(t as ProductType);
+  });
+  if (productTypes.length > 0) body.product_types = productTypes;
+
+  try {
+    const response = await axios.post(`${BASE_ENV_URL}/api/v1/serverless/projects/security`, body, {
+      headers: {
+        Authorization: `ApiKey ${apiKey}`,
+      },
+    });
+    return {
+      name: response.data.name,
+      id: response.data.id,
+      region: response.data.region_id,
+      es_url: `${response.data.endpoints.elasticsearch}:443`,
+      kb_url: `${response.data.endpoints.kibana}:443`,
+      product: response.data.type,
+    };
+  } catch (error) {
+    log.error(`${error}`);
+  }
+}
+
+// Method to invoke the delete environment API for serverless.
+async function deleteEnvironment(
+  projectId: string,
+  projectName: string,
+  apiKey: string
+): Promise<void> {
+  try {
+    await axios.delete(`${BASE_ENV_URL}/api/v1/serverless/projects/security/${projectId}`, {
+      headers: {
+        Authorization: `ApiKey ${apiKey}`,
+      },
+    });
+    log.info(`Environment ${projectName} was successfully deleted!`);
+  } catch (error) {
+    log.error(`${error}`);
+  }
+}
+
+// Method to reset the credentials for the created environment.
+async function resetCredentials(
+  environmentId: string,
+  runnerId: string,
+  apiKey: string
+): Promise<Credentials | undefined> {
+  log.info(`${runnerId} : Reseting credentials`);
+  try {
+    const response = await axios.post(
+      `${BASE_ENV_URL}/api/v1/serverless/projects/security/${environmentId}/_reset-credentials`,
+      {},
+      {
+        headers: {
+          Authorization: `ApiKey ${apiKey}`,
+        },
+      }
+    );
+    return {
+      password: response.data.password,
+      username: response.data.username,
+    };
+  } catch (error) {
+    throw new Error(`${error}`);
```

Comment:
Can you throw  `new Error(error.message)` instead of implicit `error.toString()`? It's not transparent what's `error.toString()` outputs.